src/Data/List.agda

```diff
@@ -11,7 +11,30 @@
 
 module Data.List where
 
+import Data.List.NonEmpty.Base as List⁺
+open import Function.Base using (_∘_)
+
 ------------------------------------------------------------------------
 -- Types and basic operations
 
-open import Data.List.Base public
+open import Data.List.Base public hiding (scanr)
+
+------------------------------------------------------------------------
+-- scanr
+
+module _ {a b} {A : Set a} {B : Set b} (f : A → B → B) (e : B) where
+
+  open List⁺ using (toList; scanr⁺)
+
+-- definition
+
+  scanr : List A → List B
+  scanr = toList ∘ scanr⁺ f e
```

Comment:
This is screaming to me that we're doing something wrong here with the dependencies. Can we discuss this evening at the stdlib implementor's meeting?

Reply:
Well, this *is* the motor for this PR: AFAICT, the `Data.List.Base` defn of `scanr` arises in the (historical) form it takes *precisely* to ensure that it can exist in `Data.List.Base`, without any of the other analysis of its typing. 

But, as other commenters elsewhere note, the 'official' library definition has terrible properties (or better: terrible proofs of its properties...) because of its 'wrinkly' definition, arising from its combination of `with` *and* dead-code. 

The other operations considered in #2267 / #2269 are OK, but `scanr` seems 'special'.